extension/src/pipeline/quickPick.ts

```diff
@@ -30,9 +34,37 @@ const pickDataFiles = (): Promise<string[] | undefined> =>
     'Data Formats': ['json', 'csv', 'tsv', 'yaml']
   })
 
+const formatFieldQuickPickValues = (
+  values: { file: string; key: string }[]
+) => {
+  const formattedFields: PlotConfigData['y'] = {}
+
+  for (const { file, key } of values) {
+    if (!formattedFields[file]) {
+      formattedFields[file] = key
+      continue
+    }
+
+    const prevFileValue = formattedFields[file]
+    formattedFields[file] = [
+      ...(typeof prevFileValue === 'string' ? [prevFileValue] : prevFileValue),
```

Comment:
Why not test the opposite with `Array.isArray(prevFileValue)` instead? `Array.isArray()` is better for type inference.

It also feels unnecessary to create a new array twice when `prevFileValue` is a string. Since you are creating `formattedFields` here, you can just use `push` without worrying about mutating a variable from elsewhere.